src/Tests/FlatSharpCompilerTests/FieldNameNormalizationTests.cs

```diff
@@ -60,4 +60,72 @@ struct Struct {{
             Assert.NotNull(type.GetProperty("LowerPascalCase"));
         }
     }
+
+    [Fact]
+    public void PreserveFieldCasingOnField()
+    {
+        string schema = $@"
+            {MetadataHelpers.AllAttributes}
+            namespace FieldNameNormalizationTests;
+
+            table Table {{
+                item_one : int32;
+                item_two : int32 ({MetadataKeys.PreserveFieldCasing});
+            }}
+
+            struct Struct {{
+                item_one : int32;
+                item_two : int32 ({MetadataKeys.PreserveFieldCasing});
```

Comment:
Seeing this test, this really does a bit more than just preserve casing. I think a better attribute might be `fs_preserveFieldName`, since it also keeps underscores.

Reply:
or maybe `fs_literalFieldName`?

Reply:
Or maybe just `fs_literalName`? That seems to convey the intent.

Reply:
I'll update PR to use `fs_preserveFieldName`, but am happy to change it to anything you prefer.